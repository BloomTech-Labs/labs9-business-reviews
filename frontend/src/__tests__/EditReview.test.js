import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';
import EditReviewComponent from '../components/EditReview';

describe('<EditReviewComponent />', () => {
  it('renders', () => {
    shallow(<EditReviewComponent match={{ params: '1' }} />);
  });
  it('matches snapshot', () => {
    const wrapper = shallow(<EditReviewComponent match={{ params: '1' }} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('should contain NavBar', () => {
    let app = shallow(<EditReviewComponent match={{ params: '1' }} />);
    expect(app.find("NavBar").length).toBe(1);
  });
  it('should contain 4 divs', () => {
    let app = shallow(<EditReviewComponent match={{ params: '1' }} />);
    expect(app.find("div").length).toBe(3);
  });
  it('should contain one form', () => {
    let app = shallow(<EditReviewComponent match={{ params: '1' }} />);
    expect(app.find("form").length).toBe(1);
  });
  it('should contain one h1 element with text "Edit your review"', () => {
    let app = shallow(<EditReviewComponent match={{ params: '1' }} />);
    expect(app.find("h1").length).toBe(1);
    expect(app.find("h1").text()).toMatch("Edit your review");
  });
  it('should contain 3 labels', () => {
    let app = shallow(<EditReviewComponent match={{ params: '1' }} />);
    expect(app.find("label").length).toBe(3);
  });
  it('should contain 2 buttons', () => {
    let app = shallow(<EditReviewComponent match={{ params: '1' }} />);
    expect(app.find("button").length).toBe(2);
  });
  it('should contain 5 options', () => {
    let app = shallow(<EditReviewComponent match={{ params: '1' }} />);
    expect(app.find("option").length).toBe(5);
  });
});
