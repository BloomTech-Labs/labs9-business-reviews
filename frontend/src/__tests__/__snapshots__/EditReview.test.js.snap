// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditReviewComponent /> matches snapshot 1`] = `
<div>
  <NavBar />
  <styled.div>
    <div
      className="container"
    >
      <form
        onSubmit={[Function]}
      >
        <h1
          className="review__headline"
        >
          Edit your review
        </h1>
        <label
          htmlFor="review__title"
        >
          Title
        </label>
        <input
          className="review__input--title"
          name="title"
          onChange={[Function]}
          value=""
        />
        <label
          htmlFor="review__title"
        >
          Review
        </label>
        <textarea
          className="review__input--body"
          name="body"
          onChange={[Function]}
          value=""
        />
        <div
          className="review__rating"
        >
          <label
            htmlFor="review__input--rating"
          >
            Rating
          </label>
          <select
            className="review__input--rating"
            name="rating"
            onChange={[Function]}
            value={1}
          >
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              defaultValue="5"
            >
              5
            </option>
          </select>
          <button
            className="btn"
            onClick={[Function]}
          >
            Submit Review
          </button>
          <button
            className="btn"
          >
            Cancel
          </button>
        </div>
      </form>
    </div>
  </styled.div>
</div>
`;
